diff -cr src/bluetooth.conf src/bluetooth.conf
*** src/bluetooth.conf	Wed Dec 21 17:53:54 2011
--- src/bluetooth.conf	Sat Jun 16 05:58:44 2012
***************
*** 22,27 ****
--- 22,32 ----
      <allow send_destination="org.bluez"/>
    </policy>
  
+   <!-- allow users of bluetooth group to communicate with hcid -->
+   <policy group="bluetooth">
+     <allow send_destination="org.bluez"/>
+   </policy>
+ 
    <!-- allow users of lp group (printing subsystem) to 
         communicate with bluetoothd -->
    <policy group="lp">
***************
*** 29,35 ****
    </policy>
  
    <policy context="default">
!     <deny send_destination="org.bluez"/>
    </policy>
  
  </busconfig>
--- 34,40 ----
    </policy>
  
    <policy context="default">
!     <allow send_destination="org.bluez"/>
    </policy>
  
  </busconfig>
