diff -cr audio/headset.c audio/headset.c
*** audio/headset.c	2012-03-05 10:03:57.000000000 +0100
--- audio/headset.c	2012-07-11 15:34:06.308607756 +0200
***************
*** 1208,1213 ****
--- 1208,1227 ----
  	return telephony_generic_rsp(device, CME_ERROR_NONE);
  }
  
+ static int nokia_manufacturer(struct audio_device *device, const char *buf)
+ {
+ 	struct headset *hs = device->headset;
+ 
+ 	return headset_send(hs, "Nokia\r\n");
+ }
+ 
+ static int n900_identification(struct audio_device *device, const char *buf)
+ {
+ 	struct headset *hs = device->headset;
+ 
+ 	return headset_send(hs, "N900\r\n");
+ }
+ 
  static struct event event_callbacks[] = {
  	{ "ATA", answer_call },
  	{ "ATD", dial_number },
***************
*** 1231,1236 ****
--- 1245,1252 ----
  	{ "AT+BVRA", voice_dial },
  	{ "AT+XAPL", apple_command },
  	{ "AT+IPHONEACCEV", apple_command },
+ 	{ "AT+CGMI", nokia_manufacturer },
+ 	{ "AT+CGMM", n900_identification },
  	{ 0 }
  };
  
